engine/test/engine/db/db_server.cpp

```diff
@@ -0,0 +1,1609 @@
+  vectordb::Json recordsJson;
+  EXPECT_TRUE(recordsJson.LoadFromString(records));
+  auto insertStatus = database.Insert(dbName, tableName, recordsJson);
+  EXPECT_TRUE(insertStatus.ok()) << insertStatus.message();
+  vectordb::engine::DenseVectorElement queryData[] = {0.35, 0.55, 0.47, 0.94};
+  struct TestCase {
+    std::string searchFieldName;
+    std::vector<std::string> expectedOrder;
+  };
+  std::vector<TestCase> testcases = {
+      {"EmbeddingEuclidean", {"Moscow", "Berlin", "Shanghai", "San Francisco", "London"}},
+      {"EmbeddingDotProduct", {"Moscow", "Berlin", "San Francisco", "London", "Shanghai"}},
+      {"EmbeddingCosine", {"Moscow", "Shanghai", "Berlin", "San Francisco", "London"}}};
+  auto rebuildOptions = {false, true};
+  for (auto rebuild : rebuildOptions) {
+    if (rebuild) {
+      auto rebuildStatus = database.Rebuild();
+      EXPECT_TRUE(rebuildStatus.ok()) << rebuildStatus.message();
+    }
+    for (auto &testcase : testcases) {
+      vectordb::Json result;
+      const auto limit = 6;
+      auto queryFields = std::vector<std::string>{"ID", "Doc", testcase.searchFieldName};
+      auto queryStatus = database.Search(dbName, tableName, testcase.searchFieldName, queryFields, queryDimension, queryData, limit, result, "", true);
+      EXPECT_TRUE(queryStatus.ok()) << queryStatus.message();
+      EXPECT_EQ(result.GetSize(), 5) << "duplicate insert should've been ignored";
+      for (int i = 0; i < result.GetSize(); i++) {
+        EXPECT_EQ(result.GetArrayElement(i).GetString("Doc"), testcase.expectedOrder[i])
+            << i << "th city mismatch when querying " << testcase.searchFieldName << std::endl
+            << result.DumpToString();
+      }
+    }
+  }
+}
+
+TEST(DbServer, SparseVector) {
+  std::string tempDir = std::filesystem::temp_directory_path() / std::filesystem::path("ut_db_server_sparse_vector");
+  vectordb::engine::DBServer database;
+  std::filesystem::remove_all(tempDir);
+  const auto dbName = "MyDb";
+  const auto tableName = "MyTable";
+  size_t queryDimension = 4;
+  database.LoadDB(dbName, tempDir, 150000, true);
+  size_t tableId = 0;
+
+  const std::string schema = R"_(
+{
+  "name": "MyTable",
+  "fields": [
+    {
+      "name": "ID",
+      "dataType": "INT",
+      "primaryKey": true
+    },
+    {
+      "name": "Doc",
+      "dataType": "STRING"
+    },
+    {
+      "name": "EmbeddingEuclidean",
+      "dataType": "SPARSE_VECTOR_FLOAT",
+      "dimensions": 4,
+      "metricType": "EUCLIDEAN"
+    },
+    {
+      "name": "EmbeddingDotProduct",
+      "dataType": "SPARSE_VECTOR_FLOAT",
+      "dimensions": 4,
+      "metricType": "DOT_PRODUCT"
+    },
+    {
+      "name": "EmbeddingCosine",
+      "dataType": "SPARSE_VECTOR_FLOAT",
+      "dimensions": 4,
+      "metricType": "COSINE"
+    }
+  ]
+}
+    )_";
+
+  const std::string records = R"_(
+[
+  {
+    "ID": 1,
+    "Doc": "Berlin",
+    "EmbeddingEuclidean": {
```

Comment:
Add some UT case where the sparse vector records only contain subset of indices within the dimension, and each record present different indices

Reply:
is the test in https://github.com/epsilla-cloud/vectordb/commit/46e60250f9fc9f46b929a624996ec92fa7abde89 sufficient?